symfit/core/minimizers.py

```diff
@@ -446,15 +456,87 @@ def execute(self, **minimize_options):
             **minimize_options
         )
 
+    @classmethod
+    def minimize_method(cls):
+        return "L-BFGS-B"
 
 class NelderMead(ScipyMinimize, BaseMinimizer):
     """
     Wrapper around :func:`scipy.optimize.minimize`'s NelderMead algorithm.
     """
+    @classmethod
+    def minimize_method(cls):
+        return 'Nelder-Mead'
+
     def execute(self, **minimize_options):
         return super(NelderMead, self).execute(method='Nelder-Mead', **minimize_options)
 
 
+class BasinHopping(ScipyMinimize, BaseMinimizer):
+    """
+    Wrapper around :func:`scipy.optimize.basinhopping`'s basin-hopping algorithm.
+
+    As always, the best way to use this algorithm is through
+    :class:`~symfit.core.fit.Fit`, as this will automatically select a local
+    minimizer for you depending on wether you provvided ounds, constraints, etc.
+
+    Example::
+        from symfit import Fit, parameters, cos
+        from symfit.core.minimizers import BasinHopping
+
+        x1, x2 = parameters('x1, x2')
+        model = cos(14.5 * x1 - 0.3) + (x2 + 0.2) * x2 + (x1 + 0.2) * x1
+        fit = Fit(model, minimizer=BasinHopping)
+        fit_result = fit.execute()
+
+    This will automatically compute the jacobian for you and pass it on to the
+    local minimizer, BFGS in this case.
+    """
+    @keywordonly(local_minimizer=BFGS)
+    def __init__(self, *args, **kwargs):
+        self.local_minimizer = kwargs.pop('local_minimizer')
+        super(BasinHopping, self).__init__(*args, **kwargs)
+
+        type_error = TypeError(
+            'Currently only subclasses of ScipyMinimize are supported, since'
+            ' `scipy.optimize.basinhopping` uses `scipy.optimize.minimize`.'
+        )
```

Comment:
I wouldn't do this. Make a `message` variable, and reuse that.

Reply:
good point